### PR TITLE
Updates for msprime 1.0

### PR DESCRIPTION
**Pull request by jeromekelleher**
Minimal updates to use the msprime 1.0 release, and update our testing code the use the new APIs (which generate more suitable data for us).

I hit two separate issues while doing this. I suggest we file issues for these, and merge this much @tomwhite?

There's more we can do in terms of generating cohorts more realistically, but these should probably be done in follow-ups.

### CODE REVIEW

**Review by tomwhite — COMMENTED** [review]
These changes look good @jeromekelleher. I think the new `simulate_ts` function just needs type annotations then it should be ready to go in.

**Comment by jeromekelleher** [comment]
Any idea what's happening [here](https://github.com/pystatgen/sgkit/runs/2352176242?check_suite_focus=true#step:6:24) @tomwhite? Seems like some issue with autodoc typehints? I can't reproduce locally.

**Comment by tomwhite** [comment]
I see this locally if I use sphinx-autodoc-typehints 1.12.0 (which was released earlier today), but not with 1.11.1. Perhaps we can pin that version in requirements-doc.txt.

File: sgkit/tests/test_popgen.py [hunk 1]
```diff
@@ -309,9 +345,10 @@ def test_Fst__windowed(sample_size, n_cohorts, chunks):
         )  # scikit-allel has final window missing
 
 
+@pytest.mark.skip("Bug in Tajimas D for n=100?")
```
Comment (tomwhite) [inline sgkit/tests/test_popgen.py:348]: Can we run the cases that work in the meantime?

Comment (jeromekelleher) [reply sgkit/tests/test_popgen.py:348]: Yes - I was being lazy. How do you skip a particular example with parametrize?

Comment (tomwhite) [reply sgkit/tests/test_popgen.py:348]: I don't know if it's possible. But we could remove `100` from the list, as long as we have an issue open (which we do: #523).

Comment (jeromekelleher) [reply sgkit/tests/test_popgen.py:348]: Ah, actually no, it's failing for all n > 2, so it's probably simplest to keep skipping it rather than giving a false sense of security here.

**Comment by jeromekelleher** [comment]
Hmm, still getting the failure after pinning to sphinx-autodoc-typehints-1.11.1 @tomwhite

**Comment by tomwhite** [comment]
Looks like it's this problem: https://github.com/theislab/scanpydoc/pull/32, now fixed but not released. How about using `git+https://github.com/theislab/scanpydoc.git#egg=scanpydoc` as the dependency in requirements-doc.txt? (That will need  sphinx-autodoc-typehints 1.12.0.)

File: sgkit/tests/test_popgen.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/tests/test_popgen.py]: OK, I agree it's best to skip then.

**Comment by tomwhite** [comment]
This is failing because the coverage is no longer 100% since some tests are being excluded. I had a look to see if I could see how to fix the underlying problems.
* The Fst test (`test_Fst__windowed`) passes - I couldn't reproduce the problem here. Perhaps it's OK now?
* The Tajimas D test fails. I found it's because the number of segregating sites is different: for the 10 sample test, in tskit the number of segregating sites is 31, and in sgkit it is 28. The odd thing is that there are 29 variants (sites) - so how can there be 31 segregating sites? (I'm probably missing something though!)

**Comment by tomwhite** [comment]
@jeromekelleher any thoughts on my segregating sites question above?

**Comment by jeromekelleher** [comment]
Following up now @tomwhite, thanks for the nudge

**Comment by tomwhite** [comment]
I'm using OSX - so perhaps that's causing the difference.

**Comment by jeromekelleher** [comment]
I've had a  look at the Tajima's D tests @tomwhite, and I'm not sure what's going on. I've enabled the test for n=2 so we get some coverage, so maybe we could merge this much (assuming that gives coverage)?

We're clearly not covering all the corner cases with these popgen stats currently, so I think we'll just need to follow up and beef the testing up a bit.

**Comment by jeromekelleher** [comment]
> The Tajimas D test fails. I found it's because the number of segregating sites is different: for the 10 sample test, in tskit the number of segregating sites is 31, and in sgkit it is 28. The odd thing is that there are 29 variants (sites) - so how can there be 31 segregating sites? (I'm probably missing something though!)

That's odd - can you paste in more details? (Maybe we should move this to the issue #523?)

**Review by tomwhite — APPROVED** [review]

**Comment by tomwhite** [comment]
> That's odd - can you paste in more details? (Maybe we should move this to the issue #523?)

I'll follow up in #523

**Comment by jeromekelleher** [comment]
OK, happy to merge when you are @tomwhite. I'll follow up with the Fst one now and  make a PR - that one looks easy enough to resolve.
